module/solutions/helper.js

```diff
@@ -2836,12 +2870,74 @@ module.exports = class SolutionsHelper {
 					}
 				}
 
+				if (search == undefined || search == '') search = ''
+				searchQuery = [{ title: new RegExp(search, 'i') }, { description: new RegExp(search, 'i') }]
+				let fieldsArray = [
+					'name',
+					'title',
+					'description',
+					'solutionId',
+					'programId',
+					'programInformation.name',
+					'projectTemplateId',
+					'solutionExternalId',
+					'lastDownloadedAt',
+					'hasAcceptedTAndC',
+					'referenceFrom',
+					'status',
+					'certificate',
+				]
+				if (process.env.SUBMISSION_LEVEL == 'ENTITY' && requestedData.hasOwnProperty('entityId')) {
```

Comment:
@Prajwal17Tunerlabs Like  we discussed see if we can achieve this in a single call to project collection

Reply:
optimisation of projects fetching code is achieved as discussed.